src/citrine/resources/data_concepts.py

```diff
@@ -388,6 +388,68 @@ def register_all(self, models: List[ResourceType], dry_run=False) -> List[Resour
         )
         return [self.build(obj) for obj in response_data['objects']]
 
+    def update_with_data_validation(self, model: ResourceType,
+                                    dry_run: Optional[bool] = False,
+                                    wait_for_response: Optional[bool] = True,
+                                    timeout: float = 2 * 60,
+                                    polling_delay: float = 1.0) -> Union[None, UUID]:
+        """
+        [ALPHA] Update a particular element of the collection with data validation.
+
+        Update a particular element of the collection, doing a deeper check to ensure that
+        the dependent data objects are still with the (potentially) changed constraints
+        of this change. This will allow you to make bounds and allowed named/labels changes
+        to templates.
+
+        Parameters
+        ----------
+        model: ResourceType
+            The DataConcepts object.
+        dry_run: bool
+            Whether to actually update the item or run a dry run of the update operation.
+            Dry run is intended to be used for validation. Default: false
+
+        Returns
+        -------
+        ResourceType
+            If wait_for_response if True, then this call will poll the backend, waiting
+            for the eventual job result. In the case of successful validation/update,
+            a return value of None is provided which indicates success. In the case of
+            a failure validating or processing the update, an exception (RuntimeError)
+            is raised and an error message is logged with the underlying reason of the
+            failure.
+
+            If wait_for_response if False, A job ID (of type UUID) is returned that one
+            can use to poll for the job completion and result.
```

Comment:
What public method can be used to poll the job?

Reply:
I've introduced a "poll_update_with_data_validation_job" method - which exposes separately the logic I had buried under the 'wait' flag.